lib/publiccloud/ec2.pm

```diff
@@ -36,24 +37,27 @@ sub find_img {
     return;
 }
 
+# Returns true if key is already created in EC2 otherwise tries 10 times to create it and then fails
+# If the subroutine manager to create key pair in EC2 it stores it in $self->ssh_key_pair
+
 sub create_keypair {
-    my ($self, $prefix, $out_file) = @_;
+    my ($self, $prefix) = @_;
 
-    return $self->ssh_key_file if ($self->ssh_key_file);
+    return 1 if (script_run('test -f ' . SSH_KEY_PEM) == 0);
 
     for my $i (0 .. 9) {
         my $key_name = $prefix . "_" . $i;
         my $cmd = "aws ec2 create-key-pair --key-name '" . $key_name
-          . "' --query 'KeyMaterial' --output text > " . $out_file;
+          . "' --query 'KeyMaterial' --output text > " . SSH_KEY_PEM;
         my $ret = script_run($cmd);
         if (defined($ret) && $ret == 0) {
-            assert_script_run('chmod 600 ' . $out_file);
-            $self->ssh_key($key_name);
-            $self->ssh_key_file($out_file);
-            return $key_name;
+            assert_script_run('chmod 600 ' . SSH_KEY_PEM);
+            $self->ssh_key_pair($key_name);
+            #$self->ssh_key_file($out_file);
+            return 1;
         }
```

Comment:
may be a wait sleep is useful here before next retry ?

Reply:
It was not necessary until now so I would like to keep it that way.